### PR TITLE
Fix streaming: call sink.done() on all error paths to properly terminate chunked responses

### PR DESCRIPTION
**Pull request by abn**
Without sink.done(), clients see 'incomplete chunked read' when the backend returns non-200 (e.g. context overflow) or an exception is thrown. Add sink.done() in the error branches of forward_sse_stream, forward_byte_stream, and forward_streaming_request. Add integration tests that reproduce the bug via context overflow and verify clean termination across all paths.

Relates-to: https://github.com/nowledge-co/community/issues/105

### CODE REVIEW

**Comment by bitgamma** [comment]
Looks good to me, thanks for your contribution! Please add the tests to CI. Probably is a good idea to just add them to the existing endpoint tests

File: test/server_streaming_errors.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+"""
```
Comment (bitgamma) [inline test/server_streaming_errors.py:1]: these tests need to be added to CI.

**Review by bitgamma — APPROVED** [review]
